sources/packages/web/src/services/http/RestrictionApi.ts

```diff
@@ -13,4 +19,63 @@ export class RestrictionApi extends HttpBaseClient {
     );
     return studentNotes.data as StudentRestrictionSummary[];
   }
+
+  public async getRestrictionCategories(): Promise<OptionItemDto[]> {
+    const categories = await this.getCall(
+      "restrictions/categories/options-list",
+    );
+    return categories.data as OptionItemDto[];
+  }
+
+  public async getRestrictionReasons(
+    restrictionCategory: string,
+  ): Promise<OptionItemDto[]> {
+    const reasons = await this.getCall(
+      `restrictions/reasons/options-list/category/${restrictionCategory}`,
+    );
+    return reasons.data as OptionItemDto[];
+  }
+
+  public async getStudentRestrictionDetail(
+    studentId: number,
+    studentRestrictionId: number,
+  ): Promise<StudentRestrictionDetail> {
+    const studentNotes = await this.getCall(
+      `restrictions/student/${studentId}/studentRestriction/${studentRestrictionId}`,
+    );
+    return studentNotes.data as StudentRestrictionDetail;
+  }
+
+  public async addStudentRestriction(
```

Comment:
IMO, If we are using getcall() then for consistency, we should create some functions for other API methods, or we should  go for `this.apiClient` for all  API methods